connect-button/src/main/java/com/ifttt/connect/ui/AnalyticsApiHelper.java

```diff
@@ -0,0 +1,48 @@
+package com.ifttt.connect.ui;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.Header;
+import retrofit2.http.Headers;
+import retrofit2.http.POST;
+
+final class AnalyticsApiHelper {
+
+    private static AnalyticsApiHelper INSTANCE;
+
+    private final EventsApi eventsApi;
+
+    private AnalyticsApiHelper() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        OkHttpClient okHttpClient = builder.build();
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://connect.ifttt.com")
+                .addConverterFactory(MoshiConverterFactory.create())
+                .client(okHttpClient)
+                .build();
+
+        eventsApi = retrofit.create(EventsApi.class);
+    }
+
+    static AnalyticsApiHelper get() {
```

Comment:
This method is only called by the `AnalyticsUploaderClass` once in the `doWork` block.

Reply:
Oh, and we are only going to have one thread that calls this method at any time? Because this can be called in any thread, I think we might want to make this thread-safe, i.e add `synchronize` for this method.